syntax = "proto3";

option go_package = "./pb";

message ResourceData {
  string resource_name = 1;
  float probability = 2;
  string color = 3;
}

message CreateUserRequest { int32 user_id = 1; }
message CreateUserResponse { bool created = 1; }

message GetResourceDataRequest { }
message GetResourceDataResponse { repeated ResourceData resource_dataz = 1; }

service Factory {
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {}
  rpc GetResourceData(GetResourceDataRequest) returns (GetResourceDataResponse) {}
}